presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/EvaluateArbitraryNull.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.matching.Captures;
+import com.facebook.presto.matching.Pattern;
+import com.facebook.presto.spi.plan.AggregationNode;
+import com.facebook.presto.spi.plan.Assignments;
+import com.facebook.presto.spi.plan.ProjectNode;
+import com.facebook.presto.spi.plan.ValuesNode;
+import com.facebook.presto.spi.relation.RowExpression;
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+import com.facebook.presto.sql.planner.iterative.Rule;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.common.type.UnknownType.UNKNOWN;
+import static com.facebook.presto.sql.planner.plan.Patterns.aggregation;
+import static com.facebook.presto.sql.relational.Expressions.constant;
+
+/**
+ * Evaluates arbitrary(NULL) operations to NULL.
+ * <p>
+ * From:
+ * <pre>
+ * - Aggregation (arbitrary(NULL), ...otherAggregations)
+ *   - Source
+ * </pre>
+ * To:
+ * <pre>
+ * - Project (arbitrary <- null)
+ *   - Aggregation (...otherAggregations)
+ *     - Source
+ * </pre>
+ * </p>
+ */
+public class EvaluateArbitraryNull
+        implements Rule<AggregationNode>
+{
+    private static final String ARBITRARY = "arbitrary";
+    private static final Pattern<AggregationNode> PATTERN = aggregation().matching(EvaluateArbitraryNull::hasArbitraryNull);
+
+    private static boolean hasArbitraryNull(AggregationNode aggregation)
+    {
+        return aggregation.getAggregations().values().stream()
+                .anyMatch(agg -> agg.getCall().getDisplayName().equals(ARBITRARY) && agg.getCall().getType() == UNKNOWN);
```

Comment:
This maybe inadequate - can you explicitly check there is a null literal as the argument?

Reply:
Got it. Do you have any pointers on how to retrieve the actual values of the argument, since at this stage the arguments are of type `VariableReferenceExpression`, which only has `name` and `type`.

Reply:
@kaikalur under what circumstance would we have a reference to a null literal that doesn't have the UNKNOWN type?